builder/builder-next/exporter/mobyexporter/export.go

```diff
@@ -181,8 +194,13 @@ func (e *imageExporterInstance) Export(ctx context.Context, inp *exporter.Source
 		}
 	}
 
-	return map[string]string{
+	resp := map[string]string{
 		exptypes.ExporterImageConfigDigestKey: configDigest.String(),
 		exptypes.ExporterImageDigestKey:       id.String(),
-	}, nil, nil
+	}
+	if len(names) > 0 {
+		resp["image.name"] = strings.Join(names, ",")
```

Comment:
This fixes the multiple exporters failure: https://github.com/moby/moby/pull/47364#issuecomment-1936095261

But still fails at a later stage when comparing the history event (I think we don't support it for non-containerd docker, so maybe we could skip it?)

dockerd-containerd (passing)
```
    client_test.go:2799: type:COMPLETE record:<Ref:"nxaiaaj3wi7v7gq6fu6bkn4pk" CreatedAt:<seconds:1707839156 nanos:546986594 > CompletedAt:<seconds:1707839157 nanos:332890511 > logs:<media_type:"application/vnd.buildkit.status.v0" digest:"sha256:d0b5329318f37a62d9a7ac789e5ec7a6782f15262f281746eb741d343fc778b8" size:2878 > Result:<ResultDeprecated:<media_type:"application/vnd.docker.distribution.manifest.v2+json" digest:"sha256:f5c7cdb9c351fdecbf60739e4aab24b1e248632f9afbca7c4dcbf85ce28ae8be" size:497 > Attestations:<media_type:"application/vnd.in-toto+json" digest:"sha256:d98d20ad452441f7f7ea45fdd6ea64ab1083f6d86c0ef5ccd7b11b63d3b842f9" size:2847 annotations:<key:"in-toto.io/predicate-type" value:"https://slsa.dev/provenance/v0.2" > > Results:<key:0 value:<media_type:"application/vnd.docker.distribution.manifest.v2+json" digest:"sha256:f5c7cdb9c351fdecbf60739e4aab24b1e248632f9afbca7c4dcbf85ce28ae8be" size:497 > > Results:<key:1 value:<media_type:"application/vnd.oci.image.manifest.v1+json" digest:"sha256:2c126607adf65d9fc1a3f0baaedc06ed9a6649dc5261b1bbab358f53ac721afd" size:475 > > > numTotalSteps:7 numCompletedSteps:7 >
```

dockerd-containerd (failing)
```
    client_test.go:2799: type:COMPLETE record:<Ref:"0vc6t7bgn8g3skkha6ofnb4v4" CreatedAt:<seconds:1707839154 nanos:372656260 > CompletedAt:<seconds:1707839154 nanos:396397218 > logs:<media_type:"application/vnd.buildkit.status.v0" digest:"sha256:08a151a974d7112d1fb61f84cd0231ce4585b42b698ba7210625f99ff87732e6" size:2096 > Result:<Attestations:<media_type:"application/vnd.in-toto+json" digest:"sha256:150867fe18165309af1c1648c20d2bce300b315275b849b7e5be32dd85de6d5c" size:2848 annotations:<key:"in-toto.io/predicate-type" value:"https://slsa.dev/provenance/v0.2" > > > numTotalSteps:7 numCompletedSteps:7 >
```

Reply:
Opened https://github.com/moby/buildkit/pull/4654 to fix dockerd test with multiple exporter but I'm not sure why the build result is empty in the history: https://github.com/moby/moby/actions/runs/7917802574/job/21614836923?pr=47364#step:11:3815

```
 === FAIL: client TestIntegration/TestMultipleExporters/worker=dockerd (2.16s)
    client_test.go:2749: 
        	Error Trace:	/src/client/client_test.go:2749
        	            				/src/util/testutil/integration/run.go:93
        	            				/src/util/testutil/integration/run.go:207
        	Error:      	"map[]" should have 2 item(s), but has 0
        	Test:       	TestIntegration/TestMultipleExporters/worker=dockerd
```

Might be because OCI exporter fails and therefore no result is set.

Reply:
Ok so result is still empty: https://github.com/moby/moby/actions/runs/7919334085/job/21619979340?pr=47364#step:11:3895

```
=== Failed
=== FAIL: client TestIntegration/TestMultipleExporters/worker=dockerd (2.16s)
    client_test.go:2810: 
        	Error Trace:	/src/client/client_test.go:2810
        	            				/src/util/testutil/integration/run.go:93
        	            				/src/util/testutil/integration/run.go:207
        	Error:      	"map[]" should have 1 item(s), but has 0
```

Reply:
When doing tests on my side I see `Result` is set with only the provenance and `Results` is empty:

```
client_test.go:2809: Result Attestations:<media_type:"application/vnd.in-toto+json" digest:"sha256:d1f3cb826bd745c2690ad54b3220ccc6d3ed848eb0fe6018571bd69591671b34" size:2865 annotations:<key:"in-toto.io/predicate-type" value:"https://slsa.dev/provenance/v0.2" > >
client_test.go:2810: Results map[]
```

Reply:
The only thing I see that `mobyexporter` doesn't set that `containerimage` does is:
```
resp[exptypes.ExporterImageDescriptorKey] = ... base64.StdEncoding.EncodeToString(<image descriptor>)
```

Not sure if we can produce a valid descriptor here, as we don't really store the manifest with the graphdriver store..
But maybe it's actually something different? Could you point me to the code related to this Result generation?

Reply:
I did some hacking around: https://github.com/moby/moby/pull/47364/commits/5555a658cfbb81bc440f9bd3a8cc58df513f3d6d

Let me know if that makes sense. We could also have a temporary manifest, but maybe it's fine if we just return config as result?

Reply:
Oh right it makes sense with the graph driver, completely missed that but should be good with containerd snap.

Reply:
> Let me know if that makes sense. We could also have a temporary manifest, but maybe it's fine if we just return config as result?

Yes I think that's fine, let me check that.

Reply:
Opened https://github.com/moby/buildkit/pull/4657

Reply:
Thanks! Updated this PR to use fixed test.

Reply:
Ok looks good https://github.com/moby/moby/actions/runs/7932676254/job/21659772460?pr=47364

Will check with @tonistiigi about https://github.com/moby/buildkit/pull/4657